pkg/registry/apis/provisioning/repository/local.go

```diff
@@ -356,7 +356,12 @@ func (r *localRepository) Update(ctx context.Context, path string, ref string, d
 	}
 
 	if _, err := os.Stat(path); errors.Is(err, os.ErrNotExist) {
-		return fmt.Errorf("file does not exist")
+		return &apierrors.StatusError{ErrStatus: metav1.Status{
```

Comment:
why not returning `ErrFileNotFound` here ? you translate from os.ErrNotExist to our own not found error.

Reply:
so that we get a 404 rather than a 500